typescript-test-samples/apigw-lambda-external/README.md

```diff
@@ -0,0 +1,119 @@
+[![typescript: 5.3.2](https://badgen.net/badge/Built%20With/TypeScript/blue9)](https://badgen.net/badge/Built%20With/TypeScript/blue9)
+[![test: unit](https://img.shields.io/badge/Test-Unit-blue)](https://img.shields.io/badge/Test-Unit-blue)
+[![test: integration](https://img.shields.io/badge/Test-Integration-yellow)](https://img.shields.io/badge/Test-Integration-yellow)
+
+# Typescript: Amazon Api Gateway, AWS Lambda, External API dependency
+
+## Introduction
+
+This project contains introductory examples of TypeScript tests written for AWS Lambda. This is a great place to start!
+
+The project uses the [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html) (SAM) CLI for configuration, testing and deployment.
+
+---
+
+## Contents
+
+- [Introduction](#introduction)
+- [Contents](#contents)
+- [About this Pattern](#about-this-pattern)
+- [About this Example](#about-this-example)
+  - [Key Files in the Project](#key-files-in-the-project)
+- [Sample project description](#sample-project-description)
+- [Run the Unit Test](#run-the-unit-test)
+- [Run the Integration Test](#run-the-integration-test)
+
+---
+
+## About this Pattern
+
+### System Under Test (SUT)
+
+The SUT in this pattern is a synchronous API composed of Amazon API Gateway, AWS Lambda and a simple external API.
```

Comment:
Cool idea and approach... I'd like to see a deeper description about the scenario and provided solution. Maybe in this section (or under Goal / Description) or maybe in the Introduction above... 

Explain why it is useful to have a separate mocked API for integration tests. What does it give you/why is it a good idea? What do you give up when you choose this? When would you choose to take this approach vs. other options like hitting the external API from your test environment?

Reply:
Introduced a deeper dive into the pattern including drawbacks and suggesting where the drawbacks should be catered for into the Introduction. I hope this provides more clarity.